src/ui/public/elasticsearch_errors/is_term_size_zero_error.js

```diff
@@ -1,6 +1,10 @@
 import ElasticsearchError from './elasticsearch_error';
 
 export default function isTermSizeZeroError(error) {
-  const esError = new ElasticsearchError(error);
-  return esError.hasRootCause('size must be positive, got 0');
+  try {
+    const esError = new ElasticsearchError(error);
+    return esError.hasRootCause('size must be positive, got 0');
+  } catch (err) {
+    return false;
+  }
```

Comment:
Also, could you elaborate on what you mean by unnecessarily restrictive? What would you like to be able to do, which can't be done with a class-based solution?